### PR TITLE
Feature/event

### LINKED ISSUE #21: Events
use Expressjs and Sequelize
- Routes CRUD that returns json response
- Migration structure
- Models for events.

### PR DESCRIPTION
**Pull request by rohail411**

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "development": {
-    "username": "admin",
-    "password": "admin",
+    "username": "postgres",
```
Comment (isheraz) [inline config/config.json:3]: Please add this in `.gitignore` and add an `example.config.js`

File: constants.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 const PG_HOST = process.env.PG_HOST || '127.0.0.1';
-const PG_PORT = process.env.PG_PORT || 45775;
+const PG_PORT = process.env.PG_PORT || 5432;
```
Comment (isheraz) [inline constants.js]: Please do not change the default port value and only change this in the `.env` file.

File: controllers/eventController.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+const Event = require('../models').Event;
+
+const createEvent = async (req,res)=>{
```
Comment (isheraz) [inline controllers/eventController.js]: `_req` for unused function variable. Add this rule in the `eslintrc`

File: controllers/eventController.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+const getEventById = async (req,res) => {
+    try {
+        const {id} = req.params;
+
+        const event = await Event.findOne({where:{id:id}});
+        if(event) return res.status(200).json({event});
+        throw new Error('Event Id not found');
+    } catch (error) {
+        return res.status(500).json({error});
```
Comment (isheraz) [inline controllers/eventController.js]: Define Each Error and Send Error codes with proper messages as a response. A frontend dev wouldn't understand be able to properly explain the error otherwise.

File: controllers/eventController.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+const updateEvent = async (req,res)=>{
+    try {
+        const {id} = req.params;
+        const [update] = await Event.update(req.body,{
+            where: {id:id}
+        })
+        if(update){
+            const updatedEvent = await Event.findOne({where: {id:id}});
+            return res.status(201).json({updatedEvent});
+        }else throw new Error('Something went wrong');
```
Comment (isheraz) [inline controllers/eventController.js]: What went Wrong???

File: migrations/20201223070153-event.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+'use strict';
```
Comment (isheraz) [inline migrations/20201223070153-event.js]: which migration is valid?

File: package.json [hunk 7]
```diff
@@ -40,11 +40,7 @@
     "lint-staged": "^10.5.3",
     "prettier": "2.2.1"
   },
-  "husky": {
```
Comment (isheraz) [inline package.json]: why did you remove the pre-commit?

File: routes/eventRoute.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+const eventController = require('../controllers/eventController');
```
Comment (isheraz) [inline routes/eventRoute.js:2]: Best implementation so far.

**Review by isheraz — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (rohail411) [reply package.json]: it not allow me to push

File: migrations/20201223070153-event.js (hunk 6 above)
Comment (rohail411) [reply migrations/20201223070153-event.js]: 20201223070153-event this one is valid

File: constants.js (hunk 2 above)
Comment (rohail411) [reply constants.js]: ok

File: controllers/eventController.js (hunk 4 above)
Comment (rohail411) [reply controllers/eventController.js]: readable errors added

File: config/config.json (hunk 1 above)
Comment (rohail411) [reply config/config.json:3]: added

File: migrations/20201223070153-event.js (hunk 6 above)
Comment (isheraz) [reply migrations/20201223070153-event.js]: Remove the other file.

File: controllers/eventController.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+const createEvent = async (req,res)=>{
+    try {
+        const event = await Event.create(req.body)
+        return res.status(200).json({event})
+    } catch (error) {
+        return res.status(500).json({error:'Something Went Wrong Please Try Again'})
+    }
+}
+
+const getAllEvents = async (req,res) => {
```
Comment (isheraz) [inline controllers/eventController.js]: change to const `getAllEvents = async (req,res)`

File: controllers/eventController.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    try {
+        const events = await Event.findAll();
+        return res.status(200).json({events})
```
Comment (isheraz) [inline controllers/eventController.js]: Manage no record found

File: controllers/eventController.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    } catch (error) {
+        return res.status(500).json({error:'Something Went Wrong Please Try Again'})
```
Comment (isheraz) [inline controllers/eventController.js]: Make sure your error messages are meaningful and explain the issue.

File: controllers/eventController.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+    }
+}
+
+const getEventById = async (req,res) => {
+    try {
+        const {id} = req.params;
```
Comment (isheraz) [inline controllers/eventController.js]: add validation for all your parameters.

File: controllers/eventController.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+        return res.status(404).json({error:'Event with this id not found'}); 
+    } catch (error) {
+        return res.status(500).json({error:'Something Went Wrong Please Try Again'})
+
+    }
+}
+
+const updateEvent = async (req,res)=>{
+    try {
+        const {id} = req.params;
```
Comment (isheraz) [inline controllers/eventController.js]: refer to comments above and implement for all functions in eventController.js

File: package.json (hunk 7 above)
Comment (isheraz) [reply package.json]: Do not remove husky, this will run lint before you push your code. Please refer to eslint and husky libraries and read about them. 
- Additionally, revert these

**Review by isheraz — CHANGES_REQUESTED** [review]

File: controllers/eventController.js (hunk 10 above)
Comment (rohail411) [reply controllers/eventController.js]: ok

File: migrations/20201223070153-event.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+      id:{
+          type:Sequelize.INTEGER,unique:true,primaryKey:true,autoIncrement:true
+      },
+      name:{
+          type:Sequelize.STRING,nullable:false
+      },
+      venue:{
+          type:Sequelize.STRING,nullable:false
+      },
+      date:{
```
Comment (isheraz) [inline migrations/20201223070153-event.js]: Description of event

**Review by isheraz — CHANGES_REQUESTED** [review]
